bob-backend/src/interval_logger.rs

```diff
@@ -0,0 +1,42 @@
+use coarsetime::{Duration as CDuration, Instant as CInstant};
+use std::{
+    collections::HashMap,
+    hash::Hash,
+    cmp::Eq,
+    fmt::Display
+};
+
+#[derive(Debug)]
+pub struct IntervalErrorLogger<E> {
```

Comment:
Add an additional struct `IntervalErrorLoggerSafe<E>` that wraps `IntervalErrorLogger<E>` protected by Mutex. This structure will hide the synchronization behind the scenes and thus simplify the caller's code.